cmd/gateclient/client.go

```diff
@@ -368,6 +376,53 @@ func (m *GatewayClient) authenticateIAP() (string, error) {
 	return token, err
 }
 
+func (m *GatewayClient) authenticateGoogleServiceAccount() (err error) {
+	auth := m.Config.Auth
+	gsa := auth.GoogleServiceAccount
+
+	if !gsa.IsValid() {
+		return nil
+	}
+
+	if gsa.CachedToken != nil && gsa.CachedToken.Valid() {
+		return m.login(gsa.CachedToken.AccessToken)
+	}
+	gsa.CachedToken = nil
+
+	var source oauth2.TokenSource
+	if gsa.File == "" {
+		source, err = google.DefaultTokenSource(context.Background(), "profile", "email")
+	} else {
+		p, ferr := ioutil.ReadFile(gsa.File)
+		if ferr != nil {
+			return err
+		}
+		source, err = google.JWTAccessTokenSourceFromJSON(p, "https://accounts.google.com/o/oauth2/v2/auth")
+	}
+	if err != nil {
+		return err
+	}
+
+	token, err := source.Token()
+	if err != nil {
+		return err
+	}
+
+	gsa.CachedToken = token
+	if err := m.login(token.AccessToken); err != nil {
+		return err
+	}
+	m.Context = context.Background()
+
+	// Cache token if login succeeded
+	gsa.CachedToken = token
+	if err = writeYAML(&m.Config, m.configLocation, 0600); err != nil {
```

Comment:
Same comment here about the file mode.

Reply:
Additionally if it should consistently be the same mode we can wrap if in the function until we need to refactor. Maybe specialize this as `writeConfigYAML()`.

Reply:
Did this -- so, same call everywhere.